### DISCUSSION TITLE
How can i use stylex in Next.js with swc?

### DISCUSSION THREAD

**Discussion by redish101** [facebook/stylex#644] [Q&A] [answered]
I want to use stylex in Next.js with swc,but the document only mentioned the way used babel.

**Comment by nmn** [comment]
try using the CLI.

Here's an example project: https://github.com/nmn/next-15-with-stylex-cli

**Reply by redish101** [reply]
This can complicate the build process and code structure.

**Reply by nmn** [reply, marked as answer]
Since we don't have an SWC plugin, this is the only option if you want to use SWC. It's also more reliable than how the Next.js plugin works.

Further, other solutions, like Tailwind, rely on code-gen which is essentially the same process.
